runtime/onert/backend/train/ops/PoolLayer.cc

```diff
@@ -108,6 +110,75 @@ class MaxPool2D final : public TrainingKernelRegistry
   }
 };
 
+class AveragePool2D final : public TrainingKernelRegistry
+{
+private:
+  const ir::Activation _activation;
+  const IPortableTensor *_output;
+  nnfw::cker::PoolParams _op_params;
+
+  std::unique_ptr<Tensor> _act_back_prop_output;
+  std::unique_ptr<Tensor> _arg_avg_index;
+
+public:
+  AveragePool2D(const uint32_t paddingLeft, const uint32_t, const uint32_t paddingTop,
+                const uint32_t, const uint32_t strideWidth, const uint32_t strideHeight,
+                const uint32_t kernelWidth, const uint32_t kernelHeight,
+                const ir::Activation activation, const IPortableTensor *output)
+    : _activation(activation), _output(output)
+  {
+    {
+      _op_params.stride_height = strideHeight;
+      _op_params.stride_width = strideWidth;
+      _op_params.filter_height = kernelHeight;
+      _op_params.filter_width = kernelWidth;
+      _op_params.padding_values.height = (int8_t)paddingTop;
+      _op_params.padding_values.width = (int8_t)paddingLeft;
```

Comment:
It would be better to use `static_cast` and `uint8_t`.
```suggestion
      assert(paddingTop < (1 << 8));
      assert(paddingLeft < (1 << 8));
      _op_params.padding_values.height = static_cast<uint8_t>(paddingTop);
      _op_params.padding_values.width = static_cast<uint8_t>(paddingLeft);
```

Could you also modifiy it for `MaxPool2D` in another PR?

Reply:
> Could you also modifiy it for `MaxPool2D` in another PR?

Sure, I'll do it in another PR.